report_compassion/models/contract_group.py

```diff
@@ -172,6 +172,10 @@ def get_communication(self, start, stop, sponsorships):
             elif number_sponsorship and valid.child_id:
                 vals['subject'] = valid.child_id.firstname + " ({})".format(
                     valid.child_id.local_id)
+            elif number_sponsorship and not valid.child_id and valid.display_name:
+                product_name = self.env['product.product'].search([('id', '=', valid.contract_line_ids.product_id.id)])
```

Comment:
`('id', 'in', valid.mapped('contract_line_ids.product_id').ids)`
As there can be multiple contracts in valid and multiple contract lines in one contract.